src/Chunk.cpp

```diff
@@ -335,27 +336,32 @@ void cChunk::SetAllData(cSetChunkData & a_SetChunkData)
 	}
 
 	// Clear the block entities present - either the loader / saver has better, or we'll create empty ones:
-	for (cBlockEntityList::iterator itr = m_BlockEntities.begin(); itr != m_BlockEntities.end(); ++itr)
+	for (auto & KeyPair : m_BlockEntities)
 	{
-		delete *itr;
+		delete KeyPair.second;
 	}
 	m_BlockEntities.clear();
-	std::swap(a_SetChunkData.GetBlockEntities(), m_BlockEntities);
+	for (cBlockEntity * Block : a_SetChunkData.GetBlockEntities())
+	{
+		AddBlockEntityClean(Block);
+	}
+	a_SetChunkData.GetBlockEntities().clear();
```

Comment:
I believe the `a_SetChunkData`'s block entities should use the `std::map`, too. In fact, just change the `cBlockEntityList` typedef to `std::map`, then you can keep the `swap` here. If it's not too much trouble, rename the typedef to `cBlockEntities`.